content/posts/0006-build-plan-buildpack.md

```diff
@@ -0,0 +1,109 @@
+---
+title: What is the Build Plan Buildpack and why should I care?
+date: "2021-03-05"
+slug: build-plan-buildpack-exploration
+author: feckhardt
+---
+The [Build Plan buildpack](https://github.com/paketo-community/build-plan)
+gives you very precise control over what buildpacks run during a build and what
+dependencies end up in the final image. Let's talk about how it works and
+why/when you would utilize this kind of functionality.
+
+### What is the Build Plan?
+
+A Build Plan is how a buildpack indicates the dependencies it provides and
+requires to other buildpacks. Formally, its a part of the Cloud Native
+Buildpacks (CNB) [Buildpack API
+specification](https://github.com/buildpacks/spec/blob/main/buildpack.md) and
+it functions by passing a [TOML
+file](https://github.com/buildpacks/spec/blob/main/buildpack.md#build-plan-toml)
+to the buildpack lifecycle after the detect phase has completed. There's a lot
+of nuance and complexity there but mostly this means that at the end of the
+detect phase, a buildpack will write down what it is capable of providing as a
+dependency for other buildpacks to use. It also writes down what it requires in
+order to perform its own functions.
+
+Let's show this with an example. The [`bundle-install`
+buildpack](https://github.com/paketo-buildpacks/bundle-install) performs a
+`bundle install` process which installs Ruby packages called "gems". In order
+to run `bundle install`, the buildpack needs an installation of Ruby and the
+Bundler CLI. All of this can be described in a Build Plan. The buildpack simply
```

Comment:
It'd be cool to include a snippet of TOML showing the actual plan contribution from the example buildpack.